### ISSUE TITLE
MissingMethodException with .Net Core 3.1 and EventFlow-EntityFramework-Integration

### ISSUE THREAD

**Issue by i-dentify** [eventflow/EventFlow#718]
Now that the .net core 3 PR is merged and available I tried to upgrade my current project. Unfortunately I have a very weird issue, which I cannot debug any further as I cannot get the EventFlow solution running on my MacBook (I'd debug it there, if it was running).

The exception I get is:

```
System.MissingMethodException: Method not found: 'Microsoft.EntityFrameworkCore.Metadata.Builders.IndexBuilder Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder`1.HasIndex(System.Linq.Expressions.Expression`1<System.Func`2<!0,System.Object>>)'.
   at EventFlow.EntityFramework.Extensions.ModelBuilderExtensions.AddEventFlowSnapshots(ModelBuilder modelBuilder)
```
Same goes for "AddEventFlowEvents"...

The strange thing is: if I skip these both calls and ONLY use my own model builder stuff (for my own entities), then there is no such error - though I also call ```HasIndex``` and such.
And to make it even more strange: putting the code from ```AddEventFlowEvents``` directly into my ModelBuilder-stuff does NOT throw this exception.

About the project: it uses .net core 3.1 and event flow with autofac integration and entity framework integration using postgresql. Unfortunately it is quite impossible to strip out the relevant parts as the project already got quite huge and complex. However - according to the .net core 3.0 api changes I had to do some parts different to the EventFlow examples:

AutoFac: no longer integrated within ```ConfigureServices``` but instead within ```ConfigureContainer``` being a void method.

```
public void ConfigureContainer(ContainerBuilder builder)
{
    // ...
    EventFlowOptions.New
                .Configure(options => options.IsAsynchronousSubscribersEnabled = true)
                .UseAutofacContainerBuilder(builder)
                .ConfigureEntityFramework(EntityFrameworkConfiguration.New)
                .UseHangfireJobScheduler()
                .PublishToRabbitMq(RabbitMqConfiguration.With(new Uri(this.Configuration["RabbitMQ:Url"]),
                    exchange: this.Configuration["RabbitMQ:Exchange"]))
                .UseLibLog(LibLogProviders.Serilog);
                //.CreateContainer();
} 
```
Note the point not calling ```CreateContainer()```, which threw exceptions. Obviously .net core 3 does it on its own... or at least the relevant stuff, which is done within this method.

**Comment by dennisfabri** [comment]
I am having the same problem since some time. The only workaround I have found so far, is to use EntityFrameworkCore version 2.x (2.2.6 works just fine).

**Comment by i-dentify** [comment]
Found this one: https://github.com/aspnet/EntityFrameworkCore/issues/16376

Seems, as if the EF-Integration to EventFlow should reference the new nuget package then...

**Comment by runebaekkelund** [comment]
Hi, is there any news/progress on this bug? We are in the progress of upgrading from .net core 2.2 to .net core 3.1 and this is a blocker for us.

**Comment by KasperLK** [comment]
We have temporarily resolved the issue by upgrading everything to Core 3.1, except for Entity Framework, which we have kept on version 2.2.6.
However, this causes a problem with the EF Core tools which can't find our db provider. I think it's because it is looking for CreateWebHostBuilder which has been replaced by CreateHostBuilder, so our Startup is never called.

Our workaround is to create a constructor withour parameters in our DbContext class, and set our db provider by overwriting OnConfiguring.
The workaround is only necessary when executing EF Core tools commands.

**Comment by garyevari** [comment]
@KasperLK can you please give more specifics around the resolution you were able to implement?

**Comment by KasperLK** [comment]
```
<PropertyGroup>
<TargetFramework>netcoreapp3.1</TargetFramework>
</PropertyGroup>

<ItemGroup>
<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="2.2.6" />
<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.2.6" />
<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="2.2.6">
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>
</ItemGroup>
```

In our db context class we have this code, that is commented out, unless we need to run the EF Core Tools, like add-migration etc.
```
public ReadModelDbContext()
{
}

protected override void OnConfiguring(DbContextOptionsBuilder builder)
{
    builder.UseSqlServer("");   // Insert connectionstring
    base.OnConfiguring(builder);
}
```

**Comment by rasmus** [comment]
Hi @i-dentify and @KasperLK, I create PR #742 which I guess will solve your problem. Do you agree?

**Comment by i-dentify** [comment]
> Hi @i-dentify and @KasperLK, I create PR #742 which I guess will solve your problem. Do you agree?

Yep, this should solve it.

**Comment by frankebersoll** [comment]
Moved PR to #753

**Comment by rasmus** [comment]
This has been released in 0.79.4216. A big thanks to @frankebersoll
